plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1172,6 +1174,15 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
             String fileSystem = new Path(table.getLocation()).toUri().getScheme();
             throw new TrinoException(NOT_SUPPORTED, format("Inserts are not supported on the %s filesystem", fileSystem));
         }
+        Map<String, Boolean> columnNullabilities = getColumnsNullability(table.getMetadataEntry());
+        long numberOfNonNullableColumns = columnNullabilities.values().stream().filter(nullability -> !nullability).count();
```

Comment:
I would use `anyMatch()` instead of `filter(...).count()`. The subsequent logic are also same.